flaky TestConnectionCongestionControl

```--- FAIL: TestConnectionCongestionControl (0.00s)
    connection_test.go:2256: 
        	Error Trace:	D:/a/quic-go/quic-go/connection_test.go:2256
        	Error:      	Should be true
        	Test:       	TestConnectionCongestionControl
panic: Fail in goroutine after TestConnectionCongestionControl has completed

goroutine 193 [running]:
testing.(*common).Fail(0x23fc968)
	C:/hostedtoolcache/windows/go/1.23.4/x64/src/testing/testing.go:953 +0xf1
testing.(*common).FailNow(0x23fc968)
	C:/hostedtoolcache/windows/go/1.23.4/x64/src/testing/testing.go:982 +0x2c
testing.(*common).Fatalf(0x23fc968, {0xe42e66, 0x2e}, {0x214e240, 0x5, 0x5})
	C:/hostedtoolcache/windows/go/1.23.4/x64/src/testing/testing.go:1090 +0x80
go.uber.org/mock/gomock.(*Controller).Call.func1(0x23efc80, {0xde3be0, 0x23c9190}, {0xe2319b, 0x6}, {0x2583410, 0x1, 0x1})
	C:/Users/runneradmin/go/pkg/mod/go.uber.org/mock@v0.5.0/gomock/controller.go:214 +0x363
go.uber.org/mock/gomock.(*Controller).Call(0x23efc80, {0xde3be0, 0x23c9190}, {0xe2319b, 0x6}, {0x2583410, 0x1, 0x1})
	C:/Users/runneradmin/go/pkg/mod/go.uber.org/mock@v0.5.0/gomock/controller.go:230 +0x6c
github.com/quic-go/quic-go.(*MockConnRunner).Remove(0x23c9190, {{0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...}, ...})
	D:/a/quic-go/quic-go/mock_conn_runner_test.go:158 +0xc2
github.com/quic-go/quic-go.(*connIDGenerator).RemoveAll(0x2363470)
	D:/a/quic-go/quic-go/conn_id_generator.go:122 +0x8a
github.com/quic-go/quic-go.(*connection).handleCloseError(0x24af688, 0x24ebeb4)
	D:/a/quic-go/quic-go/connection.go:1675 +0x6d4
github.com/quic-go/quic-go.(*connection).run(0x24af688)
	D:/a/quic-go/quic-go/connection.go:661 +0x116a
github.com/quic-go/quic-go.TestConnectionCongestionControl.func3()
	D:/a/quic-go/quic-go/connection_test.go:2249 +0x27
created by github.com/quic-go/quic-go.TestConnectionCongestionControl in goroutine 192
	D:/a/quic-go/quic-go/connection_test.go:2249 +0x61c

--- a/connection_test.go
+++ b/connection_test.go
@@ -69,8 +69,10 @@ func connectionOptHandshakeConfirmed() testConnectionOpt {
 	}
 }
 
-func connectionOptRTTStats(s *utils.RTTStats) testConnectionOpt {
-	return func(conn *connection) { conn.rttStats = s }
+func connectionOptRTT(rtt time.Duration) testConnectionOpt {
+	var rttStats utils.RTTStats
+	rttStats.UpdateRTT(rtt, 0)
+	return func(conn *connection) { conn.rttStats = &rttStats }
 }
 
 func connectionOptRetrySrcConnID(rcid protocol.ConnectionID) testConnectionOpt {
@@ -1584,10 +1586,6 @@ func TestConnectionPacketPacing(t *testing.T) {
 	sph := mockackhandler.NewMockSentPacketHandler(mockCtrl)
 	sender := NewMockSender(mockCtrl)
 
-	// set a fixed RTT, so that the idle timeout doesn't interfere with this test
-	var rttStats utils.RTTStats
-	rttStats.UpdateRTT(10*time.Second, 0)
-
 	tc := newServerTestConnection(t,
 		mockCtrl,
 		nil,
@@ -1595,7 +1593,8 @@ func TestConnectionPacketPacing(t *testing.T) {
 		connectionOptSentPacketHandler(sph),
 		connectionOptSender(sender),
 		connectionOptHandshakeConfirmed(),
-		connectionOptRTTStats(&rttStats),
+		// set a fixed RTT, so that the idle timeout doesn't interfere with this test
+		connectionOptRTT(10*time.Second),
 	)
 	sender.EXPECT().Run()
 
@@ -1696,8 +1695,6 @@ func TestConnectionPacketPacing(t *testing.T) {
 
 func TestConnectionIdleTimeout(t *testing.T) {
 	mockCtrl := gomock.NewController(t)
-	var rttStats utils.RTTStats
-	rttStats.UpdateRTT(time.Millisecond, 0)
 	sph := mockackhandler.NewMockSentPacketHandler(mockCtrl)
 	tc := newServerTestConnection(t,
 		mockCtrl,
@@ -1705,7 +1702,7 @@ func TestConnectionIdleTimeout(t *testing.T) {
 		false,
 		connectionOptHandshakeConfirmed(),
 		connectionOptSentPacketHandler(sph),
-		connectionOptRTTStats(&rttStats),
+		connectionOptRTT(time.Millisecond),
 	)
 	// the idle timeout is set when the transport parameters are received
 	idleTimeout := scaleDuration(50 * time.Millisecond)
@@ -1765,15 +1762,13 @@ func testConnectionKeepAlive(t *testing.T, enable, expectKeepAlive bool) {
 
 	mockCtrl := gomock.NewController(t)
 	unpacker := NewMockUnpacker(mockCtrl)
-	var rttStats utils.RTTStats
-	rttStats.UpdateRTT(time.Millisecond, 0)
 	tc := newServerTestConnection(t,
 		mockCtrl,
 		&Config{MaxIdleTimeout: time.Second, KeepAlivePeriod: keepAlivePeriod},
 		false,
 		connectionOptUnpacker(unpacker),
 		connectionOptHandshakeConfirmed(),
-		connectionOptRTTStats(&rttStats),
+		connectionOptRTT(time.Millisecond),
 	)
 	// the idle timeout is set when the transport parameters are received
 	idleTimeout := scaleDuration(50 * time.Millisecond)
@@ -1851,8 +1846,6 @@ func TestConnectionACKTimer(t *testing.T) {
 	mockCtrl := gomock.NewController(t)
 	rph := mockackhandler.NewMockReceivedPacketHandler(mockCtrl)
 	sph := mockackhandler.NewMockSentPacketHandler(mockCtrl)
-	rttStats := &utils.RTTStats{}
-	rttStats.UpdateRTT(10*time.Second, 0)
 	tc := newServerTestConnection(t,
 		mockCtrl,
 		&Config{MaxIdleTimeout: time.Second},
@@ -1860,7 +1853,7 @@ func TestConnectionACKTimer(t *testing.T) {
 		connectionOptHandshakeConfirmed(),
 		connectionOptReceivedPacketHandler(rph),
 		connectionOptSentPacketHandler(sph),
-		connectionOptRTTStats(rttStats),
+		connectionOptRTT(10*time.Second),
 	)
 	alarmTimeout := scaleDuration(50 * time.Millisecond)
 
@@ -2222,13 +2215,14 @@ func TestConnectionCongestionControl(t *testing.T) {
 		false,
 		connectionOptHandshakeConfirmed(),
 		connectionOptSentPacketHandler(sph),
+		connectionOptRTT(10*time.Second),
 	)
 
 	sph.EXPECT().TimeUntilSend().AnyTimes()
 	sph.EXPECT().GetLossDetectionTimeout().AnyTimes()
 	sph.EXPECT().ECNMode(true).AnyTimes()
 	sph.EXPECT().SendMode(gomock.Any()).Return(ackhandler.SendAny).Times(2)
-	sph.EXPECT().SendMode(gomock.Any()).Return(ackhandler.SendAck)
+	sph.EXPECT().SendMode(gomock.Any()).Return(ackhandler.SendAck).MaxTimes(1)
 	sph.EXPECT().SentPacket(gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any()).Times(2)
 	// Since we're already sending out packets, we don't expect any calls to PackAckOnlyPacket
 	for i := 0; i < 2; i++ {

